htlcswitch/iterator.go

```diff
@@ -175,15 +176,9 @@ func (r *sphinxHopIterator) ForwardingInstructions() (ForwardingInfo, error) {
 		var cid uint64
 
 		tlvStream, err := tlv.NewStream(
-			tlv.MakeDynamicRecord(
-				tlv.AmtOnionType, &amt, nil,
-				tlv.ETUint64, tlv.DTUint64,
-			),
-			tlv.MakeDynamicRecord(
-				tlv.LockTimeOnionType, &cltv, nil,
-				tlv.ETUint32, tlv.DTUint32,
-			),
-			tlv.MakePrimitiveRecord(tlv.NextHopOnionType, &cid),
+			record.NewAmtToFwdRecord(&amt),
+			record.NewLockTimeRecord(&cltv),
+			record.NewNextHopIDRecord(&cid),
```

Comment:
Is this actually partitioning the network for people running master?

Reply:
technically yes, but only for sender and receiver where one is older master and one has this commit. none of the implementations have done a release yet tho

Reply:
ok, probably not a problem then. people running master now will probably upgrade relatively often to later master commits.